urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/utils/Configuration.java

```diff
@@ -0,0 +1,343 @@
+        String type;
+        if (ext.equals("yml"))
+        {
+            type = "yaml";
+        }
+        else
+        {
+            type = "";
+        }
+        try
+        {
+            return load(new FileInputStream(file), type);
+        }
+        catch (final FileNotFoundException e)
+        {
+            e.printStackTrace();
+        }
+        return false;
+    }
+    
+    public boolean load(InputStream inputStream)
+    {
+        return load(inputStream, true);
+    }
+    
+    public boolean load(InputStream inputStream, boolean overwrite)
+    {
+        return load(inputStream, "", overwrite);
+    }
+    
+    public boolean load(InputStream inputFile, String type)
+    {
+        return load(inputFile, type, true);
+    }
+    
+    @SuppressWarnings("unchecked")
+    public boolean load(InputStream inputFile, String type, boolean overwrite)
+    {
+        if (type.equalsIgnoreCase("yaml"))
+        {
+            final Yaml yamlParser = new Yaml();
+            final Map<String, Object> data = (Map<String, Object>) yamlParser
+                    .load(inputFile);
+            if (data != null)
+            {
+                merge(data, config, overwrite);
+            }
+        }
+        else
+        {
+            final Properties props = new Properties();
+            
+            try
+            {
+                props.load(inputFile);
+            }
+            catch (final IOException e)
+            {
+                e.printStackTrace();
+                return false;
+            }
+            
+            for (final Object key : props.stringPropertyNames())
+            {
+                final String name = key.toString();
+                final String value = props.getProperty(name);
+                
+                config.put(name, value);
+            }
+        }
+        
+        return true;
+    }
+    
+    public void load(Configuration config)
+    {
+        load(config, true);
+    }
+    
+    public void load(Configuration config, boolean overwrite)
+    {
+        merge(config.getConfig(), this.config, overwrite);
+    }
+    
+    public void merge(Map<String, Object> from, Map<String, Object> to)
+    {
+        merge(from, to, true);
+    }
+    
+    @SuppressWarnings("unchecked")
+    public void merge(Map<String, Object> from, Map<String, Object> to,
+            boolean overwrite)
+    {
+        for (final String key : from.keySet())
+        {
+            if (to.get(key) == null)
+            {
+                to.put(key, from.get(key));
+            }
+            else
+            {
+                if (to.get(key) instanceof Map)
+                {
+                    merge((Map<String, Object>) from.get(key),
+                            (Map<String, Object>) to.get(key), overwrite);
+                }
+                else
+                {
+                    if (overwrite)
+                    {
+                        to.put(key, from.get(key));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.